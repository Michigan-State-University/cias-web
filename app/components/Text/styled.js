import styled from 'styled-components';

export const StyledEllipsisText = styled.div`
  position: relative;
  font-weight: bold;
  font-size: 18px;
  overflow: hidden;
  width: 100%;
  text-overflow: ellipsis;
  white-space: nowrap;
  height: 20px;
`;
