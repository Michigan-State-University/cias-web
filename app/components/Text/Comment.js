import styled from 'styled-components';
import { fontSizes, fontFamily, lineHeights, colors } from 'theme';
import Text from '.';

const Comment = styled(Text)`
  margin: 0;
  font-family: ${fontFamily};
  font-size: ${fontSizes.small};
  line-height: ${lineHeights.small};
  color: ${colors.grey};
`;

export default Comment;
