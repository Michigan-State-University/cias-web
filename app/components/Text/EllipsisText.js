import React, { useRef, useState, useEffect } from 'react';
import PropTypes from 'prop-types';

import Tooltip from 'components/Tooltip';
import Row from 'components/Row';

import { StyledEllipsisText } from './styled';

const EllipsisText = ({ text, ...props }) => {
  const ref = useRef(null);
  const [allowTooltip, setAllowTooltip] = useState(false);

  useEffect(() => {
    if (!allowTooltip && ref.current) {
      const { offsetWidth, scrollWidth } = ref.current;
      if (offsetWidth < scrollWidth) {
        setAllowTooltip(true);
      }
    }
  }, [text]);

  return (
    <Row>
      {allowTooltip ? (
        <Tooltip text={text} id={text ?? ''} width="100%" display="inline">
          <StyledEllipsisText ref={ref} {...props}>
            {text}
          </StyledEllipsisText>
        </Tooltip>
      ) : (
        <StyledEllipsisText ref={ref} {...props}>
          {text}
        </StyledEllipsisText>
      )}
    </Row>
  );
};

EllipsisText.propTypes = {
  text: PropTypes.string,
};

export default EllipsisText;
