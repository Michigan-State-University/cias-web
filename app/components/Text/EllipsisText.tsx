import React, { memo, useRef, useState } from 'react';

import Tooltip from 'components/Tooltip';
import Row from 'components/Row';

import { StyledEllipsisText } from './styled';

type EllipsisTextType = {
  text: string;
  dataFor?: string;
  lines?: number;
  width?: number;
} & Record<string, unknown>;

const EllipsisText = ({
  text,
  dataFor,
  lines,
  width,
  ...props
}: EllipsisTextType) => {
  const ref = useRef(null);

  const [allowTooltip, setAllowTooltip] = useState(false);

  const onTruncate = (isTruncated: boolean) => {
    setAllowTooltip(isTruncated);
  };

  return (
    <Row width="100%">
      <Tooltip
        visible={allowTooltip}
        text={text}
        id={text ?? ''}
        display="inline"
        width="100%"
      >
        <StyledEllipsisText
          ref={ref}
          data-tip={text}
          data-for={dataFor ?? text ?? ''}
          lines={lines}
          onTruncate={onTruncate}
          width={width}
          $styleProps={props}
        >
          {text}
        </StyledEllipsisText>
      </Tooltip>
    </Row>
  );
};

export default memo(EllipsisText);
