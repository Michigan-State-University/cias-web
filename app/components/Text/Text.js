import styled from 'styled-components';
import { fontSizes, fontFamily, lineHeights, colors } from 'theme';
import {
  margin,
  text,
  layout,
  style,
  positioning,
} from '../BaseComponentStyles';

const Text = styled.p.attrs((props) => ({
  onClick: props.disabled ? null : props.onClick,
}))`
  font-family: ${fontFamily};
  font-size: ${fontSizes.small};
  line-height: ${lineHeights.small};
  ${margin};
  ${text};
  ${layout};
  ${style};
  ${positioning};
  ${({ disabled }) =>
    disabled && { color: colors.grey, cursor: 'not-allowed' }};
`;

export default Text;
