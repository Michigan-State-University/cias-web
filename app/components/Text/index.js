import styled from 'styled-components';
import { fontSizes, fontFamily, lineHeights } from 'theme';
import { margin, text, layout, style } from '../BaseComponentStyles';

const Text = styled.p`
  font-family: ${fontFamily};
  font-size: ${fontSizes.small};
  line-height: ${lineHeights.small};
  ${margin};
  ${text};
  ${layout};
  ${style};
`;

export default Text;
