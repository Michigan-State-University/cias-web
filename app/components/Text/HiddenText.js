import styled from 'styled-components';

import Text from './Text';

const HiddenText = styled(Text)`
  clip: rect(0 0 0 0);
  clip-path: inset(100%);
  height: 1px;
  overflow: hidden;
  position: absolute;
  white-space: nowrap;
  width: 1px;
`;

export default HiddenText;
