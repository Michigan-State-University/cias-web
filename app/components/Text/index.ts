import Comment from './Comment';
import EllipsisText from './EllipsisText';
import HiddenText from './HiddenText';
import Text from './Text';

export { Text, HiddenText, EllipsisText, Comment };
export default Text;
