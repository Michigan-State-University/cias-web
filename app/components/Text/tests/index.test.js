import React from 'react';
import { render } from 'react-testing-library';
import 'jest-styled-components';

import Text from '../index';

describe('<Text />', () => {
  it('should match the snapshot', () => {
    const { container } = render(<Text />);
    expect(container).toMatchSnapshot();
  });

  it('should match the snapshot for disabled', () => {
    const { container } = render(<Text disabled />);
    expect(container).toMatchSnapshot();
  });
});
