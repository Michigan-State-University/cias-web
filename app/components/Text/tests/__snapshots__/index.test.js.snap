// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> should match the snapshot 1`] = `
.c0 {
  font-family: DM Sans;
  font-size: 13px;
  line-height: 130%;
  color: #2F3850;
}

<div>
  <p
    class="c0"
  />
</div>
`;

exports[`<Text /> should match the snapshot for disabled 1`] = `
.c0 {
  font-family: DM Sans;
  font-size: 13px;
  line-height: 130%;
  color: #2F3850;
  opacity: 0.5;
  color: #9AA1B6;
  cursor: not-allowed;
}

<div>
  <p
    class="c0"
    disabled=""
  />
</div>
`;
